src/tracks/ui/TextEditDelegate.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+class wxString;
+class wxPoint;
+
+class AudacityProject;
+
+//Used as a delegate for TextEditHelper
+class TextEditDelegate
+{
+public:
+
+    virtual ~TextEditDelegate() = default;
```

Comment:
I recommend:  Combine this header with TextEditHelper.h, because the classes cooperate closely.  Neither is useful without the other.

Then, define the destructor of TextEditDelegate as non-inline.

If a class has virtual functions, it is better that at least one of them is defined non-inline, and the destructor can be the one if all the others are pure virtuals.  (Actually pure virtual functions cal also be given definitions, but that's another story.)

It is better because then it can be decided at compile time, in what single place, the virtual function table of this base class will be compiled.

Whereas otherwise, the compiler must duplicate the virtual function table definition, and the linker needs to remove all but one duplicate.

So avoid the bit of extra work for the linker.

Reply:
> Combine this header with TextEditHelper.h, because the classes cooperate closely

They do, but it's not neccessary that owner of the TextEditHelper and TextEditDelegate are the same object, and in the latter case it's better to keep it in a separate file

Reply:
I still like non-inline virtual destructors (really, virtual anything should never be inlined, because it will usually never be inlined).